azkaban-exec-server/src/main/java/azkaban/container/FlowContainer.java

```diff
@@ -0,0 +1,336 @@
+    // Unzip the project zip
+    FlowContainer.unzipFile(projectZipPath.toString(), projectDirPath);
+
+    // TODO : Deepak - throw away after directory setup is final.
+    Path pwd = Paths.get("").toAbsolutePath();
+    System.out.println("Deepak : Dumping dir structure for = " + pwd.toString());
+    java.nio.file.Files.walk(pwd).filter(java.nio.file.Files::isRegularFile)
+        .forEach(System.out::println);
+
+    // Set Azkaban props
+    Props props = setAzkabanProps(jobtypePluginPath);
+
+    // Inject AzkabanCommonModule
+    final Injector injector = Guice.createInjector(
+        new AzkabanCommonModule(props)
+    );
+    SERVICE_PROVIDER.setInjector(injector);
+
+    // Constructor
+    final FlowContainer flowContainer =
+        new FlowContainer(projectDirPath, props);
+
+    // Use some execId to execute the flow
+    flowContainer.submitFlow(execId);
+  }
+
+  // Set Azkaban Props
+  private static Props setAzkabanProps(final Path jobtypePluginPath) {
+    final Map<String, String> propsMap = new HashMap<>();
+    propsMap.put(AzkabanExecutorServer.JOBTYPE_PLUGIN_DIR,
+        jobtypePluginPath.toString());
+
+    // Setup the azkaban.properties here.
+    final String[] args = {CONF_ARG, CONF_DIR};
+    final Props props = AzkabanServer.loadProps(args);
+
+    // Set db stuff.
+    return new Props(props, propsMap);
+  }
+
+  // Submit flow
+  public void submitFlow(final int execId) throws ExecutorManagerException {
+    final FlowRunner flowRunner = createFlowRunner(execId);
+    submitFlowRunner(flowRunner);
+  }
+
+  // create Flow Runner
+  private FlowRunner createFlowRunner(final int execId) throws ExecutorManagerException {
+    final ExecutableFlow flow = this.executorLoader.fetchExecutableFlow(execId);
+    if (flow == null) {
+      throw new ExecutorManagerException("Error loading flow with exec " + execId);
+    }
+
+    // Update the status of the flow from DISPATCHING to PREPARING
+    flow.setStatus(Status.PREPARING);
+
+    // Setup flow runner
+    FlowWatcher watcher = null;
+    final ExecutionOptions options = flow.getExecutionOptions();
+    if (options.getPipelineExecutionId() != null) {
+      final int pipelinedExecId = options.getPipelineExecutionId();
+      watcher = new RemoteFlowWatcher(pipelinedExecId, this.executorLoader);
+    }
+
+    // TODO : figure out the metrics
+    final MetricsManager metricsManager = new MetricsManager(new MetricRegistry());
+    final CommonMetrics commonMetrics = new CommonMetrics(metricsManager);
+    final ExecMetrics execMetrics = new ExecMetrics(metricsManager);
+    final FlowRunner flowRunner = new FlowRunner(flow, this.executorLoader,
+        this.projectLoader, this.jobTypeManager, this.azKabanProps, null,
+            null, commonMetrics, execMetrics);
+    flowRunner.setFlowWatcher(watcher)
+        .setJobLogSettings(this.jobLogChunkSize, this.jobLogNumFiles)
+        .setValidateProxyUser(this.validateProxyUser)
+        .setNumJobThreads(20);
```

Comment:
Please create constant for numJobThreads. Isn't there any property for this value?

Reply:
will do.